tools/buildtools/schc.py

```diff
@@ -0,0 +1,923 @@
+
+        assert tokenProperties.macro is not None, labeledExpr
+        currentMacro = f"    /* 0x{currentOffset:0{offsetWidth}X} */ {tokenProperties.macro}("
+
+        if tokenProperties.isUnconditionalBranch:
+            targetOffset = getTargetOffset(labeledExpr, labeledList, offsetList)
+            diff = targetOffset - nextOffset
+            if tokenProperties.isShort:
+                if diff not in range(-0x80, 0x7F):
+                    fatalError(f"Trying to use a short command, but the branch distance is too big to fit on a single byte ({diff})", labeledExpr.token.filename, labeledExpr.token.lineNumber, labeledExpr.token.columnNumber)
+            else:
+                if diff not in range(-0x8000, 0x7FFF):
+                    fatalError(f"Trying to use a long command, but the branch distance is too big to fit on a single byte ({diff})", labeledExpr.token.filename, labeledExpr.token.lineNumber, labeledExpr.token.columnNumber)
+            currentMacro += f"0x{targetOffset:0{offsetWidth}X} - 0x{nextOffset:0{offsetWidth}X}"
+        else:
+            if labeledExpr.args is not None:
+                currentMacro += f"{labeledExpr.args.tokenLiteral}"
+            if tokenProperties.isConditionalBranch:
+                targetOffset = getTargetOffset(labeledExpr, labeledList, offsetList)
+                diff = targetOffset - nextOffset
+                if tokenProperties.isShort:
+                    if diff not in range(-0x80, 0x7F):
+                        fatalError(f"Trying to use a short command, but the branch distance is too big to fit on a single byte ({diff})", labeledExpr.token.filename, labeledExpr.token.lineNumber, labeledExpr.token.columnNumber)
+                else:
+                    if diff not in range(-0x8000, 0x7FFF):
+                        fatalError(f"Trying to use a long command, but the branch distance is too big to fit on a single byte ({diff})", labeledExpr.token.filename, labeledExpr.token.lineNumber, labeledExpr.token.columnNumber)
+                currentMacro += f", 0x{targetOffset:0{offsetWidth}X} - 0x{nextOffset:0{offsetWidth}X}"
+        currentMacro += "),"
+
+        if debuggingLevel >= 1:
+            currentMacro += f" /* {labeledExpr.token.filename}:{labeledExpr.token.lineNumber}:{labeledExpr.token.columnNumber} */"
+
+        currentMacro += "\n"
+
+        result += currentMacro
+
+    return result
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Compiler for the high level schedule language", prog=__prog_name__)
+
+    parser.add_argument("input", help="Schedule script path", type=Path)
+    parser.add_argument("-o", "--output", help="Output path. Will print to stdout if omitted", type=Path)
+    parser.add_argument("-g", type=int, nargs="?", const=1, default=0, dest="debuggingLevel", metavar="level", help="Emit debugging information on the generated macros. Level 0 means no debugging information. Passing no level at all implies level 1. Defaults to level 0")
```

Comment:
```suggestion
    parser.add_argument("-g", type=int, nargs="?", const=1, default=0, dest="debuggingLevel", metavar="level", help="Emit debugging information on the generated macros. Level 0 means no debugging information. -g is like -g1. Default is -g0")
```
Is a lot less confusing to me if I got it right